compiler/luci/service/src/Nodes/CircleReshape.cpp

```diff
@@ -95,21 +97,13 @@ loco::TensorShape Algorithm::visit(const luci::CircleReshape *node)
     }
     else
     {
-      // We use shape from the node itself
-      loco::TensorShape shape;
-      shape.rank(node->rank());
-      for (uint32_t r = 0; r < node->rank(); ++r)
+      auto shape = loco::must_cast<luci::CircleNode *>(node->shape());
+      shape_by_input.rank(shape->dim(0).value());
```

Comment:
```suggestion
      auto shape_node = loco::must_cast<luci::CircleNode *>(node->shape());
      assert(shape_node->rank() == 1);
      // shape_node tensor values will provide new shape, like [2, 3, 4]
      auto num_elements = shape_node->dim(0).value(); // above example will give 3
      shape_by_input.rank(num_elements);
```

Reply:
Thanks for your kind suggestion.
With this in mind, I will make my variable names clear and easy to understand.